openshift-knative-operator/pkg/serving/extension.go

```diff
@@ -164,6 +168,14 @@ func (e *extension) Reconcile(ctx context.Context, comp base.KComponent) error {
 		common.ConfigureIfUnset(&ks.Spec.CommonSpec, monitoring.ObservabilityCMName, monitoring.ObservabilityBackendKey, "none")
 	}
 
+	cleanDomainMappingOnce.Do(func() {
```

Comment:
@skonto would it cover the delete once idea?

Reply:
Yes that is the idea. If the pod is restarted it will re-run but that is ok.

Reply:
https://github.com/openshift-knative/serverless-operator/pull/2743/commits/cf5317c47b4d7d8551ed9839059f5a065c5003a6

Reply:
Thinking again this it seems that we cannot use the error returned so users will have to restart the pod, not ideal. So better use a mutex or preferably`atomic.Bool ` (I think we dont have more than one reconciliations happening at the same time) and if an error happens it can be returned so a new reconciliation will happen. If there is no error we just the atomic.bool and we never re-run it.